### PR TITLE
Feature/apl 635 node status with api

### PR DESCRIPTION
**Pull request by oleksiylukin**
It is not fully ready yet, this PR is just for developers co-operation

### CODE REVIEW

**Comment by oleksiylukin** [comment]
PR is ready please review

File: apl-api/src/main/java/com/apollocurrency/aplwallet/api/dto/DurableTaskInfo.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Schema(name="Name of task", description="Short but descruiptive name of task")
+    public String name;    
+    @Schema(name="Description of task", description="Description of task in one line")
+    public String decription;
+    @Schema(name="Is task critical", description="Taks is critical and should be displayed on top of UI window")
+    public Boolean isCrititcal = false;
+    @Schema(name="Task state", description="Task state in one line in human readable form", 
+            allowableValues = {"Starded","In progress","Finished","Cancelled","Paused"})
+    public String stateOfTask;
+    @Schema(name="Task stafrt date", description="Task start date and time")
```
Comment (AndrewBoyarsky) [inline apl-api/src/main/java/com/apollocurrency/aplwallet/api/dto/DurableTaskInfo.java]: fix typo 'stafrt'->'start'

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplAppStatus.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+ * @author alukin@gmail.com
+ */
+@Singleton
+public class AplAppStatus {
+    private static final Logger LOG = getLogger(AplAppStatus.class);
+    private static final long ONE_DAY=3600*24;
+    private final Map<String,DurableTaskInfo> tasks = new HashMap<>();
+    private NumberFormat formatter = new DecimalFormat("#000.000");
+    
+    @Inject
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplAppStatus.java]: Nothing to inject here

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplCore.java [hunk 3]
```diff
@@ -220,18 +229,26 @@ private void startUp() {
...
                 AddOns.init();
-                AppStatus.getInstance().update("API initialization...");
                 Helper2FA.init(databaseManager);
+                aplAppStatus.durableTaskUpdate(initCoreTaskID,  70.1, "Apollo cor cleaases initialization done");
 //signal to API that core is reaqdy to serve requests. Should be removed as soon as all API will be on RestEasy                
                 ApiSplitFilter.isCoreReady = true;
 
-                ThreadPool.scheduleThread("Active connections logger", () -> LOG.debug("Used connections - '{}'", databaseManager.getDataSource().getJmxBean().getActiveConnections()), 15, TimeUnit.SECONDS);
+                ThreadPool.scheduleThread("DB_con_log_AplAppStatus_clean", 
+                   new Runnable() {
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplCore.java]: Better to use lamda expression 
`ThreadPool.scheduleThread("Active connections logger", () -> {
LOG.debug("Used connections - '{}'", databaseManager.getDataSource().getJmxBean().getActiveConnections());
aplAppStatus.clearFinished(10*60L); //10 min
}
, 15, TimeUnit.SECONDS)`

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/config/PropertyBasedFileConfig.java [hunk 4]
```diff
@@ -11,22 +11,23 @@
...
 import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.CDI;
 import javax.inject.Inject;
 import javax.inject.Named;
 
 public class PropertyBasedFileConfig {
     private static final Logger LOG = LoggerFactory.getLogger(PropertyBasedFileConfig.class);
     private final PropertiesHolder propertiesHolder;
-
+    private static final AplCoreRuntime aplCoreRuntime = CDI.current().select(AplCoreRuntime.class).get();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/config/PropertyBasedFileConfig.java]: Make it non-static, (required for stateless multiple weld container init in unit tests )

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/migrator/ApplicationDataMigrationManager.java [hunk 5]
```diff
@@ -22,6 +23,7 @@
 public class ApplicationDataMigrationManager {
 
     private static final Logger LOG = LoggerFactory.getLogger(ApplicationDataMigrationManager.class);
+    private static final AplCoreRuntime aplCoreRuntime = CDI.current().select(AplCoreRuntime.class).get();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/migrator/ApplicationDataMigrationManager.java]: Make it non-static

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/BackendControlController.java [hunk 6]
```diff
@@ -44,26 +47,64 @@ public BackendControlController(BackendControlService bcService) {
...
             }
     )
-    public Response getBackendStatus(@QueryParam("detailed") @DefaultValue("false") Boolean detailed) {
-        BackendStatusResponse statusResponse = new BackendStatusResponse();
-        statusResponse.message = "Seems that server is OK :) TODO: fill with resl data";
-        statusResponse.backendInfo = bcService.getStatus();
-        if (detailed) {
-            statusResponse.message += " Details: shit happens sometimes.";
-        }
+    public Response getBackendStatus(@QueryParam("status") @DefaultValue("All") String state) {
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/BackendControlController.java]: Is it a public endpoint? Will it secure enough?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplAppStatus.java (hunk 2 above)
Comment (oleksiylukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplAppStatus.java]: Well, yes but I'd rather indicate that this is injectable entity

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/migrator/ApplicationDataMigrationManager.java (hunk 5 above)
Comment (oleksiylukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/migrator/ApplicationDataMigrationManager.java]: fixed

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplCore.java (hunk 3 above)
Comment (oleksiylukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplCore.java]: It is matter of taste, my is explicit Runnable

**Pull request merged**